models/user.ts

```diff
@@ -116,6 +116,21 @@ export const throwIfNotAllowed = (
   );
 };
 
+export const throwIfNotOwnerOnOwner = (
+  teamMemberRole: Role,
+  memberRole: Role
+) => {
+  if (
+    (teamMemberRole as Role) === 'ADMIN' &&
+    (memberRole as Role) === 'OWNER'
+  ) {
+    throw new ApiError(
+      403,
+      `Operation not allowed: An admin cannot tamper with the team owner.`
+    );
+  }
+};
+
```

Comment:
code is repeated, we already have the same logic in `validateUpdateRole` function